akka-sample-kafka-to-sharding-scala/processor/src/main/scala/sample/sharding/kafka/UserGrpcService.scala

```diff
@@ -2,25 +2,27 @@ package sample.sharding.kafka
 
 import akka.actor.typed.ActorRef
 import akka.actor.typed.ActorSystem
+import akka.actor.typed.delivery.ConsumerController.SequencedMessage
 import akka.actor.typed.scaladsl.AskPattern._
 import akka.util.Timeout
 import sample.sharding.kafka.UserEvents.GetRunningTotal
 import sample.sharding.kafka.UserEvents.RunningTotal
 
 import scala.concurrent.Future
 import scala.concurrent.duration._
+import akka.cluster.sharding.typed.ShardingEnvelope
 
 class UserGrpcService(system: ActorSystem[_]) extends UserService {
 
   implicit val timeout = Timeout(5.seconds)
   implicit val sched = system.scheduler
   implicit val ec = system.executionContext
 
-  private val shardRegion: ActorRef[UserEvents.UserQuery] = UserEvents.querySide(system)
-
-  override def userStats(in: UserStatsRequest): Future[UserStatsResponse] = {
-    shardRegion
-      .ask[RunningTotal](replyTo => GetRunningTotal(in.id, replyTo))
-      .map(runningTotal => UserStatsResponse(in.id, runningTotal.totalPurchases, runningTotal.amountSpent))
+ override def userStats(in: UserStatsRequest): Future[UserStatsResponse] = {
+   // shardRegion
```

Comment:
This  is the bit i find ticky once the sharded actor is typed to reliable delivery